src/redis.ts

```diff
@@ -0,0 +1,1256 @@
+    return this._commandParser(command, collection, key, start, stop);
+  }
+
+  zremrangebyscore(
+    key: string,
+    minimum: string,
+    maxiumum: string,
+    collection: string
+  ) {
+    const command: string = "ZREMRANGEBYSCORE";
+    return this._commandParser(command, collection, key, minimum, maxiumum);
+  }
+
+  zrevrange(
+    key: string,
+    start: string,
+    stop: string,
+    collection: string,
+    withScores: boolean = false
+  ) {
+    const command: string = "ZREVRANGE";
+    let withScoresCommand: string | undefined;
+    if (withScores) {
+      withScoresCommand = "WITHSCORES";
+    } else {
+      withScoresCommand = undefined;
+    }
+    return this._commandParser(
+      command,
+      collection,
+      key,
+      start,
+      stop,
+      withScoresCommand
+    );
+  }
+
+  zrevrangebylex(
+    key: string,
+    min: string,
+    max: string,
+    collection: string,
+    offset: string | undefined = undefined,
+    count: number | undefined = undefined
+  ) {
+    const command: string = "ZREVRANGEBYLEX";
+
+    let limitList: any[] = [];
+    if (offset !== undefined && count !== undefined) {
+      limitList.push("LIMIT");
+      limitList.push(offset);
+      limitList.push(count);
+    }
+    return this._commandParser(
+      command,
+      collection,
+      key,
+      min,
+      max,
+      ...limitList
+    );
+  }
+
+  zrevrangebyscore(
+    key: string,
+    min: string,
+    max: string,
+    collection: string,
+    withScores: boolean = false,
+    offset: string | undefined = undefined,
+    count: number | undefined = undefined
+  ) {
+    const command: string = "ZREVRANGEBYSCORE";
+    let withScoresCommand: string | undefined;
+    if (withScores) {
+      withScoresCommand = "WITHSCORES";
+    } else {
+      withScoresCommand = undefined;
+    }
+    let limitList: any[] = [];
+    if (offset !== undefined && count !== undefined) {
+      limitList.push("LIMIT");
+      limitList.push(offset);
+      limitList.push(count);
+    }
+    return this._commandParser(
+      command,
+      collection,
+      key,
+      min,
+      max,
+      withScoresCommand,
+      ...limitList
+    );
+  }
+
+  zrevrank(key: string, member: string, collection: string) {
+    const command: string = "ZREVRANK";
+    return this._commandParser(command, collection, key, member);
+  }
+
+  zscan(
+    key: string,
+    cursor: number,
+    collection: string,
+    pattern?: string | undefined,
+    count?: number | undefined
+  ) {
+    const command: string = "ZSCAN";
+
+    let patternArray: any[] = [];
+    if (pattern !== undefined) {
+      patternArray.push("MATCH");
+      patternArray.push(pattern);
+    }
+
+    let countArray: any[] = [];
+    if (count !== undefined) {
+      countArray.push("COUNT");
+      countArray.push(count);
+    }
+
+    return this._commandParser(
+      command,
+      collection,
+      key,
+      cursor,
+      ...patternArray,
+      ...countArray
+    );
```

Comment:
We can make ```hscan, zscan and sscan ``` functions inner content common passing command separately. basically common function for ``` line no 975 to 994.```

Reply:
I agree, I will do the refactoring. 👍